### ISSUE TITLE
Integrating GOT-OCR2.0 in Transformers 🤗

### ISSUE THREAD

**Issue by yonigozlan** [Ucas-HaoranWei/GOT-OCR2.0#137]
Hi!
First of all, congrats on such a great model!
I am an MLE at Hugging Face, and given the popularity and performance of your model, we are eager to integrate it into the Transformers 🤗 library. If you are interested in working with us (mostly helping us debugging if needed or clarifying certain aspects of the model) that would be great! 
Looking forward to hearing back from you

Best,
Yoni

**Comment by Ucas-HaoranWei** [comment]
Hi Yoni,
It's an honor to integrate GOT into Transformers. 
If you need any help, please feel free to contact me anytime.
My email is `weihaoran18@mails.ucas.ac.cn`

Best，
Haoran Wei

**Comment by yonigozlan** [comment]
Hi again Haoran,
The PR is up [here](https://github.com/huggingface/transformers/pull/34721) :).
For now, I intend to only support inference in Transformers. In your experience since the model came out, are there a lot of users fine-tuning this model? I am guessing maybe at least for the stage3 post-training as desribed in the paper?
I am trying to gauge how useful it would be to support this last stage of fine-tuning in post-training in Transformers. In any case, it can always be added later.
Thank you!

**Comment by Ucas-HaoranWei** [comment]
> Hi again Haoran, The PR is up [here](https://github.com/huggingface/transformers/pull/34721) :). For now, I intend to only support inference in Transformers. In your experience since the model came out, are there a lot of users fine-tuning this model? I am guessing maybe at least for the stage3 post-training as described in the paper? I am trying to gauge how useful it would be to support this last stage of post-training in Transformers. In any case, it can always be added later. Thank you!

Hi, Yoni,
Great job, I will review this PR as soon as possible.
There are still quite a few users who need to fine-tune the model, but I also suggest that you wait and see what people's future requirements are and then decide whether to incorporate training and fine-tuning later.
Thank you.

**Comment by yonigozlan** [comment]
Hi Haoran!
The PR is almost ready to go! I just wanted to ask some clarification on the box finegrained option. I saw that formats such as [x, y] or [x, y, x, y] could be added to the queries, but I was wondering what the x and y precisely are in both cases. Are they the coordinates in image pixels? In resized image pixels? And I'm confused in the first case how you can define a bow with two coordinates. Thank you in advance!

**Comment by Ucas-HaoranWei** [comment]
Hi Yoni,
Thank you for your work
[x y] is a single point, which can be ignored because GOT does not use the single-point positioning text (it is an initial idea that is not adopted later). [x0 y0 x1 y1] is a box, x0 y0 is the top left corner, and x1 y1 is the bottom right corner. They  are the coordinates in image pixels.   Their definition method is first to normalize the coordinates and then by 1000 times to eliminate decimal points.

**Comment by yonigozlan** [comment]
Thanks for the explanation!
By the way, would you like us to transfer the weights of the transformers implementation to your hub organization page? And if so what would you like us to name the weights? I was thinking of stepfun-ai/GOT-OCR2.0-hf , as we usually use the suffix "-hf" to signal a transformers model.

**Comment by Ucas-HaoranWei** [comment]
Of course,  '-hf' is fine.

**Comment by yonigozlan** [comment]
Hi @Ucas-HaoranWei ! So sorry for the delay, there was a long backlog of reviews before getting to this one, but this time it's approved and ready to go! One last thing would be if you could invite me to the stepfun-ai org on Hugging Face (username yonigozlan), so that I can transfer the weights to `stepfun-ai/GOT-OCR2.0-hf`

**Comment by Ucas-HaoranWei** [comment]
> Hi [@Ucas-HaoranWei](https://github.com/Ucas-HaoranWei) ! So sorry for the delay, there was a long backlog of reviews before getting to this one, but this time it's approved and ready to go! One last thing would be if you could invite me to the stepfun-ai org on Hugging Face (username yonigozlan), so that I can transfer the weights to `stepfun-ai/GOT-OCR2.0-hf`

Hi Yoni,
Great work! I have invited you to stepfun-ai org. The invited link: https://huggingface.co/organizations/stepfun-ai/join/ElDUhkgougwyzUnV

**Comment by yonigozlan** [comment]
Thanks! I moved the checkpoints, and GOT-OCR2.0 is now merged in the main branch of Transformers!
I'll probably communicate on it at the beginning of next week to have more traction, but of course feel free to communicate on this if and whenever you want :) 

Here are the checkpoints: [stepfun-ai/GOT-OCR-2.0-hf](https://huggingface.co/stepfun-ai/GOT-OCR-2.0-hf)
Also adapted a HF Space using the Transformers implementation: https://huggingface.co/spaces/yonigozlan/GOT-OCR-Transformers

**Comment by yonigozlan** [comment]
Communicated about the integration in Transformers on [LinkedIn](https://www.linkedin.com/posts/yoni-gozlan_got-ocr-20-the-powerful-ocr-model-from-ugcPost-7292232087040851968-tjo7?utm_source=share&utm_medium=member_desktop) :). 

Btw, I've [opened a PR](https://huggingface.co/stepfun-ai/GOT-OCR2_0/discussions/40) on your checkpoint on the hub to remove the Transformers tag, as it could be confusing for users. Do you mind merging it?

Thanks!

**Comment by Ucas-HaoranWei** [comment]
No problem, I have merged your pr.

**Comment by yonigozlan** [comment]
Hey again @Ucas-HaoranWei ! We were wondering if GOT was indeed finetuned on matplotlib data, and if it was possible to use it to output matploltlib code from an image.
Also, would you be open to communicating on slack? It would also be great for future collaborations with stepfun ;).
If so, could you send me your email at yoni.gozlan@huggingface.co so that I can invite you?
Thanks!

**Comment by Ucas-HaoranWei** [comment]
Hi, Yoni
I don't train GOT on image-matplotlib data pairs. We can use corresponding data to fine-tune the current model if there are special requirements.
I'm more than happy to communicate. My email is _weihaoran@megvii.com_.
Best
